src/riscv.h

```diff
@@ -207,25 +227,82 @@ void rv_halt(riscv_t *rv);
 /* return the halt state */
 bool rv_has_halted(riscv_t *rv);
 
-/* return the flag of outputting exit code */
-bool rv_enables_to_output_exit_code(riscv_t *rv);
+typedef struct {
+    char *elf_program;
+} vm_user_t;
+
+typedef struct {
```

Comment:
Let's hold off on including the system emulation components until we are fully prepared to merge them.